src/libsystemd/sd-bus/bus-introspect.c

```diff
@@ -101,10 +101,10 @@ static int introspect_write_arguments(struct introspect *i, const char *signatur
 
                 fprintf(i->f, "   <arg type=\"%.*s\"", (int) l, signature);
 
-                if(*names) {
-                    int ret = fprintf(i->f, " name=\"%s\"", names)-8;
-                    if(ret>0)
-                        names += ret+1;
+                if (*names) {
+                    int len = fprintf(i->f, " name=\"%s\"", names)-8;
+                    if (len > 0)
```

Comment:
i think if we number of names in the nulstr doesn't match the number of arguments we expect, then this should be caught early on (i.e. when a vtable is registered to sd-bus), as that's a programming error (and not a runtime one). And we should add debug logging here about this.

Reply:
ok, I am adding this check, but then I won't add anything here because it is already validated.

One more note is that I would also apply the member_name_is_valid() to each name, even if I see no constraints on the dbus specification for the member variable names. And this to avoid a trivial injection attach with a name like:
        "name\"> <a ref=\"bogussite\"> <tag=\"name"
this can potentially hurt an application reading the dbus interfaces and parsing automatically.

what do you think?

Reply:
yes, doing the validation at time of add makes sense